.github/workflows/bin-image.yml

```diff
@@ -0,0 +1,184 @@
+name: bin-image
+
+# Default to 'contents: read', which grants actions to read commits.
+#
+# If any permission is set, any permission not included in the list is
+# implicitly set to "none".
+#
+# see https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - 'master'
+    tags:
+      - 'v*'
+  pull_request:
+
+env:
+  REPO_SLUG: moby/vpnkit-bin
+jobs:
+  prepare:
+    runs-on: ubuntu-24.04
+    timeout-minutes: 20 # guardrails timeout for the whole job
```

Comment:
Is 20 minutes enough for this one? I recall you mentioned building the image here was rather slow

Reply:
It's only the prepare step. Build has bigger timeout:
https://github.com/moby/vpnkit/blob/207eb9fbe7d9b74d6e05f1481cca0ca6cfcc9c8a/.github/workflows/bin-image.yml#L71

OTOH, it could even be lower (like 60 minutes).

The workflow is much faster now (it took ~10 minutes in the private repo) due to using the native github runner:

https://github.com/moby/vpnkit/blob/207eb9fbe7d9b74d6e05f1481cca0ca6cfcc9c8a/.github/workflows/bin-image.yml#L70